### PR TITLE
GH-758: Admin page for Announcements management

### PR DESCRIPTION
**Pull request by tamarayustian**

### CODE REVIEW

File: ui/src/components/admin/announcements/AdminAnnouncementContainer.js [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+          }
+          if (item.startDate) {
+            return DateTime.fromISO(item.startDate) < today;
+          }
+          return false;
+        });
+        setAnnouncementList(past);
+      }
+    } catch (err) {
+      console.log(err);
```
Comment (albertparedandan) [inline ui/src/components/admin/announcements/AdminAnnouncementContainer.js:74]: use your toast

File: ui/src/components/admin/announcements/AdminAnnouncementContainer.js [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        </Button>
+      </Stack>
+      <Heading as="h2" size="lg" pt={7}>
+        {isCurrentAnnouncements
+          ? 'Current Announcements'
+          : 'Past Announcements'}
+      </Heading>
+      <List spacing="2" pt={3}>
+        {/* List announcements */}
+        {announcementList.map((announcementItem) => (
```
Comment (albertparedandan) [inline ui/src/components/admin/announcements/AdminAnnouncementContainer.js:253]: woudl it be better to use agGrid to show these announcements? feel like it's easier to edit that way, and format will be much more similar to what we get from google docs

**Review by albertparedandan — CHANGES_REQUESTED** [review]

File: ui/src/components/admin/announcements/AdminAnnouncementContainer.js (hunk 2 above)
Comment (tamarayustian) [reply ui/src/components/admin/announcements/AdminAnnouncementContainer.js:253]: agGrid not as user friendly cuz we may extend it to non-tech people for delegation - Andrea, CXO

File: ui/src/components/admin/announcements/AdminAnnouncementContainer.js [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+          }
+          return false;
+        });
+        setAnnouncementList(past);
+      }
+    } catch (err) {
+      console.log(err);
+      toast({
+        description:
+          'There was an issue with the request, please talk to a t3ch support',
```
Comment (bhryck) [inline ui/src/components/admin/announcements/AdminAnnouncementContainer.js:77]: lol

File: ui/src/components/admin/announcements/AdminAnnouncementContainer.js [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+                      )}
+                    </Text>
+                  </Grid>
+                </Stack>
+                <Spacer />
+                {/* Buttons to publish, edit, duplicate, delete */}
+                <HStack spacing={1}>
+                  {announcementItem.isInWeb && isCurrentAnnouncements && (
+                    <Button
+                      colorScheme="purple"
```
Comment (bhryck) [inline ui/src/components/admin/announcements/AdminAnnouncementContainer.js:341]: minor ux request: can we do a diff color for wether its published or not to more easily see

File: ui/src/components/admin/announcements/AdminAnnouncementContainer.js [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+  Stack,
+  HStack,
+  List,
+  ListItem,
+  Flex,
+  Spacer,
+  Badge,
+  Image,
+  Grid,
+  GridItem,
```
Comment (bhryck) [inline ui/src/components/admin/announcements/AdminAnnouncementContainer.js:19]: gk kepake

File: ui/src/components/MainContainer.js [hunk 6]
```diff
@@ -265,13 +266,19 @@ const MainContainer = () => {
           permissions={['t3ch', 'admin', 'stewardship']}
           component={FormManager}
         />
-        <PrivateRoute 
+        <PrivateRoute
+          exact
+          path="/admin/announcements"
+          permissions={['t3ch', 'admin', 'stewardship']}
```
Comment (bhryck) [inline ui/src/components/MainContainer.js:272]: iirc we wanted vp to have readonly access to this page. should we create a new user group, or is that not in scope of this pr

Comment (tamarayustian) [reply ui/src/components/MainContainer.js:272]: yea, i have a new issue for that PR. i think it gets too big to review if I combine everything here. wanna make your life easier too hehe

File: ui/src/components/admin/announcements/AdminAnnouncementContainer.js (hunk 4 above)
Comment (tamarayustian) [reply ui/src/components/admin/announcements/AdminAnnouncementContainer.js:341]: ok i changed the bg color

**Review by bhryck — APPROVED** [review]

**Review by albertparedandan — APPROVED** [review]

**Pull request merged**